src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -34,43 +37,72 @@
      * @return the command based on the user input
      * @throws ParseException if the user input does not conform the expected format
      */
-    public Command parseCommand(String userInput) throws ParseException {
+    public Command parseCommand(String userInput, CommandBox commandBox) throws ParseException {
         final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput.trim());
-        if (!matcher.matches()) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
-        }
+        final Matcher dynamicMatcher = DYNAMIC_COMMAND_FORMAT.matcher(userInput.trim());
+
+        if (dynamicMatcher.matches() && dynamicMatcher.group("inputType").equals("dynamic")) {
+            final String arguments = dynamicMatcher.group("arguments").trim();
+            String[] splitArgs = arguments.split(" ");
+
+            if (isCommand(arguments, splitArgs)) {
+                commandBox.setDynamicAsFalse();
+                return new ListCommand();
+            } else {
+                commandBox.setDynamicAsTrue();
+                return new FindCommandParser().parse(arguments);
+            }
+        } else {
```

Comment:
Is it better to move the `else` block outside (i.e. delete `else {` and `}`)?

Reply:
I think it's okay because the else block essentially deals with the case when the user enters a command.